src/Angor/Client/Services/FounderKeyService.cs

```diff
@@ -0,0 +1,80 @@
+using Angor.Shared.Models;       
+using Angor.Shared.Services;     
+using Nostr.Client.Messages;     
+using Angor.Client.Storage;
+
+public class FounderKeyService : IFounderKeyService
+{
+    private readonly IWalletStorage _walletStorage;
+    private readonly IRelayService _relayService;
+    private readonly ISerializer _serializer;
+    
+
+
+
+    public FounderKeyService(
+        IWalletStorage walletStorage,
+        IIndexerService indexerService,
+        IRelayService relayService,
+        IClientStorage storage,
+        ISerializer serializer)
+    {
+        _walletStorage = walletStorage;
+        _relayService = relayService;
+        _serializer = serializer;
+    }
+
+    public async Task<bool> IsFounderKeyInUseAsync(string founderKey)
+    {
+        var keys = _walletStorage.GetFounderKeys();
```

Comment:
I'm not sure the relay is the right place to check, we need to check with the indexer if the transaction exists.